mavedb_mapping/transcript_selection.py

```diff
@@ -0,0 +1,319 @@
+import requests
+import nest_asyncio
+import asyncio
+from cool_seq_tool.data_sources.mane_transcript_mappings import MANETranscriptMappings
+from cool_seq_tool.data_sources.uta_database import UTADatabase
+from Bio.Seq import Seq
+from bs4 import BeautifulSoup
+from mavedb_mapping.transcript_selection_helper import HelperFunctionsForBLATOutput
+from mavedb_mapping import sr, dp
+
+utadb = UTADatabase()
+mane = MANETranscriptMappings()
+
+
+nest_asyncio.apply()
+
+
+async def mapq(locs: list, chrom: str):
+    """
+    Runs a query on UTADB to obtain transcripts
+    Parameters
+    ----------
+        locs: list
+           Locations list
+
+        chrom: str
+            Chromosome
+
+        gsymb: str
+            Gene symbol.
+
+
+    Returns:
+    --------
+        list:
+            List of transcripts from UTADB
+
+    """
+    transcript_lists = []
+    for i in range(len(locs)):
+        testquery = f"""select *
+                            from uta_20210129.tx_exon_aln_v
+                            where {locs[i][0]} between alt_start_i and alt_end_i
+                            or {locs[i][1]} between alt_start_i and alt_end_i
+                            and alt_ac = '{chrom}'"""
+
+        out = await utadb.execute_query(testquery)
+        tl = []
+        for j in range(len(out)):
+            if out[j]["tx_ac"].startswith("NR_") == False:
+                tl.append(out[j]["tx_ac"])
+        if tl != []:
+            transcript_lists.append(tl)
+    return transcript_lists
+
+def using_uniprot(uniprot: str, ts: str):
+    """
+    Looks for transcripts using Uniprot ID
+
+    Parameters
+    ----------
+        uniprot: str
+            Uniprot ID
+
+        ts: str
+            Target Sequence
+
+
+    Returns:
+    --------
+        start: str
+
+        full_match:bool
+
+    """
+    try:
+        url = "https://www.uniprot.org/uniprot/" + uniprot + ".xml"
+        page = requests.get(url)
+        page = BeautifulSoup(page.text)
+        page = page.find_all("sequence")
+        up = page[1].get_text()
+        stri = str(ts)
+        if up.find(stri) != -1:
+            full_match = True
+        else:
+            full_match = False
+        start = up.find(stri[:10])
+        return start, full_match
+    except:
+        #no transcripts foundś
+        return "NA", "NA"
```

Comment:
For production-ready code, it's good practice to avoid blanket `except` statements (the first half of [this post](https://stackoverflow.com/a/21553825/7417073) is a great explanation). Ideally, we should know the kinds of exceptions that we think are possible, and narrowly catch them in specific places so that we can take specific actions in response.

So, some pseudocode might look like the following:

```
url = "https://www.uniprot.org/uniprot/" + uniprot + ".xml"
response = requests.get(url)
if response.status_code != 200:
    return "NA", "NA"
parsed = BeautifulSoup(response.text)
sequences = parsed.find_all("sequence")
try:
    top = page[1].get_text
except (IndexError, AttributeError):
    return "NA", "NA"
# etc etc
```
    # etc etc
except